api/tests/acceptance/application/snapshot-controller_test.js

```diff
@@ -0,0 +1,136 @@
+      name: 'competence-name-1',
+      index: '1.1',
+      areaId: 'recAreaA',
+      level: -1,
+      courseId: 'recBxPAuEPlTgt72q11'
+    },
+    {
+      id: 'recCompB',
+      name: 'competence-name-2',
+      index: '1.2',
+      areaId: 'recAreaB',
+      level: -1,
+      courseId: 'recBxPAuEPlTgt72q99'
+    }],
+    areas: [{ id: 'recAreaA', name: 'domaine-name-1' }, { id: 'recAreaB', name: 'domaine-name-2' }],
+    organizations: []
+  };
+
+  const inserted_user = {
+    firstName: faker.name.firstName(),
+    lastName: faker.name.lastName(),
+    email: faker.internet.email(),
+    password: userPassword,
+    cgu: true
+  };
+
+  const inserted_organization = {
+    name: 'The name of the organization',
+    email: 'organization@email.com',
+    type: 'PRO'
+  };
+
+  before(() => {
+    return knex.migrate.latest()
+      .then(() => {
+        return knex.seed.run();
+      }).then(() => {
+        return knex('users').insert(inserted_user);
+      }).then((result) => {
+        userId = result.shift();
+        inserted_organization['userId'] = userId;
+        return knex('organizations').insert(inserted_organization);
+      }).then((organization) => {
+        organizationId = organization.shift();
+      });
+  });
+
+  after(function(done) {
+    server.stop(done);
+  });
+
+  describe('POST /api/snapshots', function() {
+
+    let payload;
+    let options;
+    let injectPromise;
+
+    beforeEach(() => {
+      payload = {
+        data: {
+          attributes: {
+            'organization-id': organizationId
+          }
+        }
+      };
+
+      options = {
+        method: 'POST',
+        url: '/api/snapshots',
+        payload
+      };
+
+      options['headers'] = { authorization: 'VALID_TOKEN' };
+      sinon.stub(authorizationToken, 'verify').resolves(userId);
+      sinon.stub(profileService, 'getByUserId').resolves(fakeBuildedProfile);
+      injectPromise = server.inject(options);
+    });
+
+    afterEach(() => {
+      authorizationToken.verify.restore();
+      profileService.getByUserId.restore();
+      return knex('snapshots').delete();
+    });
+
+    it('should return 201 HTTP status code', () => {
+      // When
+      return injectPromise.then((response) => {
+        // then
+        expect(response.statusCode).to.equal(201);
+        expect(response.result.data.id).to.exist;
+      });
+    });
+
+    describe('when creating with a wrong payload', () => {
+
+      it('should return 400 HTTP status code', () => {
+        // Given
+        payload.data.attributes['organization-id'] = null;
+
+        // Then
+        const creatingSnapshotWithWrongParams = server.inject(options);
+
+        // Then
+        return creatingSnapshotWithWrongParams.then((response) => {
+          const parsedResponse = JSON.parse(response.payload);
+          expect(parsedResponse.errors[0].detail).to.equal('Cette organisation n’existe pas');
+          expect(response.statusCode).to.equal(400);
```

Comment:
Si c'est un problème sur la validité des données, alors ce n'est pas une `400` mais une `422`.

cf. https://www.bennadel.com/blog/2434-http-status-codes-for-invalid-data-400-vs-422.htm